superset-frontend/src/views/CRUD/data/database/DatabaseModal/index.tsx

```diff
@@ -1346,14 +1357,33 @@ const DatabaseModal: FunctionComponent<DatabaseModalProps> = ({
           payload: { login_method: method },
         })
       }
-      removeSSHTunnelConfig={() =>
-        setDB({
-          type: ActionType.removeSSHTunnelConfig,
-        })
-      }
     />
   );
 
+  const renderDefaultSSHTunnelSwitch = () => (
```

Comment:
@Antonio-RiveroMartnez one improvement we could consider is to converting renderDefaultSSHTunnelSwitch function into an actual component in separate file.  Then import it at top of file.  Also add a prop which is `isSSHTunneling` where the switch component just returns undefined so the show / hide logic is internal to the switch.  Then we can collapse the conditional rendering blocks into a simple normal component block.

add the isSSHTunneling to interface:
````
export interface SwitchProps {
  isEditMode: boolean;
  isSSHTunneling: boolean;
  dbFetched: any;
  disableSSHTunnelingForEngine?: boolean;
  useSSHTunneling: boolean;
  setUseSSHTunneling: React.Dispatch<React.SetStateAction<boolean>>;
  setDB: React.Dispatch<any>;
}
```

Then where we set the SSHTunnelSwitchExtension we could change to something like:

```
import SSHTunnelSwitch from './SSHTTunnelSwitch';
const SSHTunnelSwitchComponent= extensionsRegistry.get(
    'ssh_tunnel.form.switch',
  ) ?? SSHTunnelSwitch;
```

this way lower in the code on lines 1591-1601 and 1877-1897
can look like:
```
<SSHTunnelSwitchComponent
  isEditMode={isEditMode}
  isSSHTunneling={isSSHTunneling}
  dbFetched={dbFetched}
  disableSSHTunnelingForEngine={disableSSHTunnelingForEngine}
  useSSHTunneling={useSSHTunneling}
  setUseSSHTunneling={setUseSSHTunneling}
  setDB={setDB}
/>

```
and not need to do all the condition checks if using extension vs base component, etc.  what do you think?

Reply:
Oh! Yeah, that makes sense. Let me work on that and update the PR. Will come back to you when ready 👍 Thanks!